Mathlib/FieldTheory/AlgebraicClosure.lean

```diff
@@ -0,0 +1,198 @@
+/-
+Copyright (c) 2024 Jiedong Jiang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu, Jiedong Jiang
+-/
+import Mathlib.FieldTheory.NormalClosure
+import Mathlib.FieldTheory.IsAlgClosed.Basic
+import Mathlib.FieldTheory.IntermediateField.Algebraic
+
+/-!
+# Relative Algebraic Closure
+
+In this file we construct the relative algebraic closure of a field extension.
+
+## Main Definitions
+
+- `algebraicClosure F E` is the relative algebraic closure (i.e. the maximal algebraic subextension)
+  of the field extension `E / F`, which is defined to be the integral closure of `F` in `E`.
+
+-/
+noncomputable section
+
+open Polynomial FiniteDimensional IntermediateField Field
+
+variable (F E : Type*) [Field F] [Field E] [Algebra F E]
+variable {K : Type*} [Field K] [Algebra F K]
+
+/--
+The relative algebraic closure of a field `F` in an extension `E`, or called maximal
+algebraic subextension
+of `E / F`, is defined to be the subalgebra `integralClosure F E`
+upgraded to an intermediate field (when `F` and `E` are both fields). This is exactly the
+intermediate field of `E / F` consisting of all integral/algebraic elements.
+-/
+def algebraicClosure : IntermediateField F E :=
```

Comment:
Should this live in the root namespace? Then the name seems ambiguous, because it might also be an absolute algebraic closure...

Reply:
I think as @alreadydone said, this is aligned with the current naming convention. See also [normalClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/NormalClosure.html#normalClosure) Do you suggest we should change all [separableClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/SeparableClosure.html#separableClosure), algebraicClosure and [separableClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/SeparableClosure.html#separableClosure) in to `IntemediateField.fooClosure` version? I would be happy to do this in another PR if necessary.